src/edu/harvard/CS50.java

```diff
@@ -256,9 +258,13 @@ public static long getLong(String prompt)
      */
     public static String getString(String prompt)
     {
+
+        // instantiate scanner once
+        if (s == null)
+            s = new Scanner(System.in);
+
         // try to get a String from user, returning null on error
         System.out.print(prompt);
-        Scanner s = new Scanner(System.in).useDelimiter("\\n|\\r|\\r\\n");
```

Comment:
Do we not want to preserve `useDelimeter`, since we do, per C library, want to read full line, not just up until whitespace? Cf. https://docs.oracle.com/javase/7/docs/api/java/util/Scanner.html#default-delimiter.